File: 1047-maximize-sum-of-array-after-k-negations/maximize-sum-of-array-after-k-negations.cpp
class Solution {
public:
    int largestSumAfterKNegations(vector<int>& v, int k) {
        sort(v.begin(), v.end());
        for(int i=0;i<v.size() && v[i]<0 && k>0 ;i++){
            v[i]*=-1;
            k--;
        }
        int min=INT_MAX;
        int sum=0;
        for(int i=0;i<v.size();i++){
            min=std::min(min,v[i]);
            sum+=v[i];
        }
        if(k%2!=0) sum-=min*2;
        return sum;
    }
};

File: 1047-maximize-sum-of-array-after-k-negations/maximize-sum-of-array-after-k-negations.py
class Solution:
    def largestSumAfterKNegations(self, v: List[int], k: int) -> int:
        v.sort()
        for i in range(len(v)):
            if v[i] < 0 and k > 0:
                v[i] *= -1
                k -= 1
        min_val = min(v)
        sum_val = sum(v)
        if k % 2 != 0:
            sum_val -= min_val * 2
        return sum_val